### PR TITLE
bigquery: implement regionalization

### PR DESCRIPTION
**Pull request by pongad**

### CODE REVIEW

File: google-cloud-bigquery/src/main/java/com/google/cloud/bigquery/spi/v2/BigQueryRpc.java [hunk 1]
```diff
@@ -29,6 +30,7 @@
 import com.google.cloud.bigquery.BigQueryException;
 import java.util.Map;
 
+@InternalExtensionOnly
```
Comment (pongad) [inline google-cloud-bigquery/src/main/java/com/google/cloud/bigquery/spi/v2/BigQueryRpc.java]: This is technically breaking change to a public interface, but I don't think we can do anything if we don't do this :(

File: google-cloud-bigquery/src/main/java/com/google/cloud/bigquery/BigQuery.java [hunk 2]
```diff
@@ -1176,4 +1176,7 @@ TableResult query(QueryJobConfiguration configuration, JobId jobId, JobOption...
    * @throws BigQueryException upon failure
    */
   TableDataWriteChannel writer(WriteChannelConfiguration writeChannelConfiguration);
+
+  // TODO(pongad): document
+  TableDataWriteChannel writer(JobId jobId, WriteChannelConfiguration writeChannelConfiguration);
```
Comment (tswast) [inline google-cloud-bigquery/src/main/java/com/google/cloud/bigquery/BigQuery.java]: 👍 

I assume you'll document before merging?

**Comment by tswast** [comment]
We should follow this PR with another change that adds location to BigQueryOptions or something (a peer to where the default project is set).

File: google-cloud-bigquery/src/main/java/com/google/cloud/bigquery/BigQuery.java (hunk 2 above)
Comment (pongad) [reply google-cloud-bigquery/src/main/java/com/google/cloud/bigquery/BigQuery.java]: Will do this also in a follow up

**Pull request merged**